docs/proposals/005_carbon_loading_mechanism.md

```diff
@@ -0,0 +1,252 @@
+  "xyz": "extensions/xyz/index.js"
+}
+```
+
+Additionally, this setup supports extensions hosted on CDNs. Versioning can be specified either directly in the URL path or through an additional attribute:
+
+```json
+{
+  "xyz": "https://cdn.com/xyz"
+}
+```
+
+`iFrames` are also supported by placing an extension in the extensions/xyz directory and loading it in index.js.
+
+**Internal Extensions**: These must be built first and then bundled together with the shell application. Greenhouse uses Turbo's dependency resolution to ensure that each extension is built before the shell application built. Additionally, the extensions are included as dependencies in the shell application's package.json file, ensuring they are bundled seamlessly with the shell application.
+
+**External Extensions**: These need to be retrieved, built, and tested during the build process. Once ready, they are added as dependencies in the shell application's package.json file, ensuring they are seamlessly bundled with the shell application.
+
+### Diagram
+
+```bash
+                      Build Process
+  ------------------------------------------------
+  |                                              |
+  |          +-----------------------+           |
+  |          | Internal Extensions   |           |
+  |          | (Built First)         |           |
+  |          +-----------------------+           |
+  |                                              |
+  |          +-----------------------+           |
+  |          | External Extensions   |           |
+  |          | (Retrieved, Built,    |           |
+  |          | and Tested)           |           |
+  |          +-----------------------+           |
+  |                                              |
+  ------------------------------------------------
+                      |
+                      v
+           +--------------------------+
+           |  Generate Manifest File  |
+           | (Defines Available       |
+           | Extensions)              |
+           +--------------------------+
+                      |
+                      v
+     +---------------------------------------+
+     | Shell Application                     |
+     | - Bundled Extensions                  |
+     | - References Manifest for Dynamic     |
+     |   Loading at Runtime                  |
+     +---------------------------------------+
+                      |
+                      v
+        Dynamic Loading of Extensions at Runtime
+
+```
+
+### Key Concepts
+
+- Co-located extensions are bundled together with the shell application.
+- Extensions are loaded dynamically at runtime.
+- A manifest file is used to define the extensions that can be loaded.
+- Internal extensions are built first and bundled together with the shell application.
+- External extensions are retrieved, built, tested and bundled with the shell application.
+- Extensions can be hosted on CDNs.
+- Extensions can be loaded in `iFrames`.
```

Comment:
If every extension is part of the bundle then it means it can be loaded lazily using `import` whereas an iFrame take a `url` as source which in this case will not be available.

Reply:
I am not sure about this. @andypf proposed in an email following:
```
  iFrames: Supported by placing an extension in extensions/xyz and loading it in index.js.
```

Reply:
For handling iFrames, we could create a local extension and manage everything related to the iFrame within this extension. This way, the iFrame can be loaded and handled in a modular and isolated manner.

Reply:
I agree with @andypf's suggestion to modularize this implementation. I’ve added a description outlining the technical details of how an extension could be loaded as an iframe. Let me know, @andypf and @taymoor89, if this approach makes sense to you.
https://github.com/cloudoperators/juno/pull/639/files#diff-03e53d69c5bd4e2dc1a27ec01307b2ab6062070afb8895839546edf1ffb8d2cfR57